share/functions/funced.fish

```diff
@@ -102,14 +102,28 @@ function funced --description 'Edit function definition'
     else
         echo $init >$tmpname
     end
+
     # Repeatedly edit until it either parses successfully, or the user cancels
     # If the editor command itself fails, we assume the user cancelled or the file
     # could not be edited, and we do not try again
     while true
+        if which md5sum > /dev/null
```

Comment:
In practice this will only be run once but it's still better to move it outside the `while` loop. Also, all vars local to the function should be declared as such. Otherwise you risk modifying a global var by the same name. That is,

```
set -l checksum
if type -q md5sum
    set checksum (md5sum $tmpname)
end
```

Note tht it's more idiomatic to use `if type -q md5sum` which will do the right thing if md5sum is a function and doesn't require the redirection. Below you'll need to modify the test to read `if set -q checksum[1]`.